### PR TITLE
add arches query data prep

### PR DESCRIPTION
**Pull request by adamgayoso**
This adds a new static method to the `ArchesMixin` that will pad and subset query data appropriately given a reference model (instance or path). Arches works well if not too many 0 genes are padded, so a warning is issued if too many are added.

### CODE REVIEW

File: scvi/model/base/_archesmixin.py [hunk 1]
```diff
@@ -229,3 +300,17 @@ def requires_grad(key):
             par.requires_grad = True
         else:
             par.requires_grad = False
+
+
+def _get_loaded_data(reference_model, device=None):
```
Comment (jjhong922) [inline scvi/model/base/_archesmixin.py:305]: typings

File: scvi/model/base/_archesmixin.py [hunk 2]
```diff
@@ -155,6 +154,78 @@ def load_query_data(
...
+
+        This function will return a new AnnData object with padded zeros
+        for missing features, as well as correctly sorted features.
+
+        Parameters
+        ----------
+        adata
+            AnnData organized in the same way as data used to train model.
+            It is not necessary to run setup_anndata,
+            as AnnData is validated against the saved `scvi` setup dictionary.
```
Comment (jjhong922) [inline scvi/model/base/_archesmixin.py:175]: Fix comment, refer to the registry instead (same from where it was copied from)

File: scvi/model/base/_archesmixin.py [hunk 3]
```diff
@@ -155,6 +154,78 @@ def load_query_data(
...
+        _, var_names, _ = _get_loaded_data(reference_model)
+        var_names = pd.Index(var_names)
+
+        if return_reference_var_names:
+            return var_names
+
+        intersection = adata.var_names.intersection(var_names)
+        inter_len = len(intersection)
+        if inter_len == 0:
+            msg = "No reference var names found in query data. "
```
Comment (jjhong922) [inline scvi/model/base/_archesmixin.py:198]: just wrap in () to stretch across lines

Comment (jjhong922) [reply scvi/model/base/_archesmixin.py:198]: e.g.
```
raise ValueError("No reference ..."
"etc etc"
)
```

File: scvi/model/base/_archesmixin.py [hunk 4]
```diff
@@ -155,6 +154,78 @@ def load_query_data(
...
+
+        Returns
+        -------
+        Query adata ready to use in `load_query_data` unless `return_reference_var_names`
+        in which case a pd.Index of reference var names is returned.
+        """
+        _, var_names, _ = _get_loaded_data(reference_model)
+        var_names = pd.Index(var_names)
+
+        if return_reference_var_names:
```
Comment (jjhong922) [inline scvi/model/base/_archesmixin.py:192]: this is kind of strange sub sub functionality of this function. Just split into its own public function instead?

File: scvi/model/base/_archesmixin.py [hunk 5]
```diff
@@ -155,6 +154,78 @@ def load_query_data(
...
+            msg = "No reference var names found in query data. "
+            msg += "Please rerun with return_reference_var_names=True "
+            msg += "to see reference var names."
+            raise ValueError(msg)
+
+        ratio = inter_len / len(var_names)
+        logger.info("Found {}% reference vars in query data.".format(ratio * 100))
+        if ratio < 0.8:
+            warnings.warn(
+                "Query data contains less than 80% of reference var names. "
```
Comment (jjhong922) [inline scvi/model/base/_archesmixin.py:207]: nit: could make the comment a function of the ratio `f'{ratio_min:.0f}%'`

File: scvi/model/base/_archesmixin.py [hunk 6]
```diff
@@ -155,6 +154,78 @@ def load_query_data(
...
+            [adata, adata_padding],
+            axis=1,
+            join="outer",
+            index_unique=None,
+            merge="unique",
+        )
+        adata_out._inplace_subset_var(var_names)
+
+        if inplace:
+            adata._init_as_actual(adata_out, dtype=adata._X.dtype)
```
Comment (jjhong922) [inline scvi/model/base/_archesmixin.py:225]: if we start using this a lot, we might want to wrap it in a util since its not guaranteed stable

**Review by jjhong922 — APPROVED** [review]
mostly nits

File: scvi/model/base/_archesmixin.py (hunk 4 above)
Comment (adamgayoso) [reply scvi/model/base/_archesmixin.py:199]: I know... I didn't want to add another method for something so basic...

File: scvi/model/base/_archesmixin.py (hunk 3 above)
Comment (adamgayoso) [reply scvi/model/base/_archesmixin.py:205]: I'm terrible at long strings...

File: scvi/model/base/_archesmixin.py (hunk 6 above)
Comment (adamgayoso) [reply scvi/model/base/_archesmixin.py:232]: agree, though now as part of scverse we can do what we want! right?!

File: scvi/model/base/_archesmixin.py (hunk 4 above)
Comment (adamgayoso) [reply scvi/model/base/_archesmixin.py:199]: though could just be a method called `load_var_names`

File: scvi/model/base/_archesmixin.py (hunk 6 above)
Comment (jjhong922) [reply scvi/model/base/_archesmixin.py:232]: 🤔

**Review by jjhong922 — APPROVED** [review]

**Pull request merged**